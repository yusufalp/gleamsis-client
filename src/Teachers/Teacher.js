import React from 'react';
import { Link } from 'react-router-dom';
import DataContext from '../DataContext';

class Teacher extends React.Component {
  static contextType = DataContext;
  state = {
    confirm: false,
    message: ''
  }
  handleDeleteTeacher(teacher) {
    this.setState({
      confirm: true,
      message: `BE CAREFUL. Deleting a teacher will delete all the courses this teacher is assigned. Do you want to delete teacher ${teacher.firstName} ${teacher.lastName}?`
    })
  }
  handleConfirmTeacher = (teacher) => {
    this.context.deleteTeacher(teacher)
    this.setState({
      confirm: false
    })
  }
  handleCancelTeacher = () => {
    this.setState({
      confirm: false
    })
  }

  render() {
    let teacher = this.context.teachers.find(teacher => teacher.id === Number(this.props.match.params.id)) || {}
    let courses = this.context.courses.filter(course => course.teacher_id === teacher.id)
    return (
      <div>
        <h2>Hello admin,</h2>
        <p>This is {teacher.first_name} {teacher.last_name}'s information</p>
        <p>First Name: {teacher.first_name}</p>
        <p>Last Name: {teacher.last_name}</p>
        <p>Courses:</p>
        {courses.map((course, i) =>
          <p key={i}><Link to={`/courses/${course.id}`}>{course.name}</Link></p>
        )}
        <button
          className={`btn-teacher ${teacher.id === 0 ? "hidden" : ""}`}
          onClick={e => this.handleDeleteTeacher(teacher)}>Delete Teacher</button>
        <div className={`${this.state.confirm ? "" : "hidden"}`}>
          <p>{this.state.message}</p>
          <button onClick={e => this.handleConfirmTeacher(teacher)}>Yes</button>
          <button onClick={this.handleCancelTeacher}>No</button>
        </div>
      </div>
    )
  }
}

export default Teacher;