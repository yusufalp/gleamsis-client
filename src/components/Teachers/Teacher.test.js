import React from 'react';
import ReactDOM from 'react-dom';
import Teacher from './Teacher';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

describe('Teacher component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Teacher />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Teacher />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})