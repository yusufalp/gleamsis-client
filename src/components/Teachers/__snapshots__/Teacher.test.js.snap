// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teacher component renders the UI as expected 1`] = `
<div
  className="dash"
>
  <h2>
    Hello admin,
  </h2>
  <div
    className="card-display"
  >
    <p>
      This is 
       
      's information
    </p>
    <p>
      First Name: 
    </p>
    <p>
      Last Name: 
    </p>
  </div>
  <div
    className="card-display"
  >
    <p>
      Courses teaching:
    </p>
  </div>
  <button
    className=""
    onClick={[Function]}
    type="submit"
  >
    Delete Teacher
  </button>
  <div
    className="delete-message hidden"
  >
    <p>
      
    </p>
    <button
      className="btn-delete"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      onClick={[Function]}
    >
      No
    </button>
  </div>
</div>
`;
